Source/KakapoDB.swift

```diff
@@ -31,6 +42,33 @@ private final class ArrayBox<T> {
     private var value: [T]
 }
 
+private extension Array {
+    
+    func indexOf<T: Storable>(element: T) -> Int? {
```

Comment:
What about not extending for this and using:
```swift
array.indexOf { (element) -> Bool in
    return ...
}
```
or if used in multiple places and you prefer to extend Array then having a fun in KakapoDB please name it differently because this is kind of messing with `public func indexOf(element: Self.Generator.Element) -> Self.Index?` (swift is same and doesn't override but will be hard to distinguish the two)